pkg/services/ngalert/notifier/channels/teams.go

```diff
@@ -1,31 +1,208 @@
 package channels
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
+	"fmt"
 
 	"github.com/pkg/errors"
 	"github.com/prometheus/alertmanager/template"
 	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
 
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/models"
 	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/notifications"
 )
 
-// TeamsNotifier is responsible for sending
-// alert notifications to Microsoft teams.
-type TeamsNotifier struct {
-	*Base
-	URL          string
-	Message      string
-	Title        string
-	SectionTitle string
-	tmpl         *template.Template
-	log          log.Logger
-	ns           notifications.WebhookSender
-	images       ImageStore
+const (
+	ImageSizeSmall  = "small"
+	ImageSizeMedium = "medium"
+	ImageSizeLarge  = "large"
+
+	TextColorDark      = "dark"
+	TextColorLight     = "light"
+	TextColorAccent    = "accent"
+	TextColorGood      = "good"
+	TextColorWarning   = "warning"
+	TextColorAttention = "attention"
+
+	TextSizeSmall      = "small"
+	TextSizeMedium     = "medium"
+	TextSizeLarge      = "large"
+	TextSizeExtraLarge = "extraLarge"
+	TextSizeDefault    = "default"
+
+	TextWeightLighter = "lighter"
+	TextWeightBolder  = "bolder"
+	TextWeightDefault = "default"
+)
+
+// AdaptiveCardsMessage represents a message for adaptive cards.
+type AdaptiveCardsMessage struct {
+	Attachments []AdaptiveCardsAttachment `json:"attachments"`
+	Type        string                    `json:"type"`
+}
+
+// NewAdaptiveCardsMessage returns a message prepared for adaptive cards.
+// https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using#send-adaptive-cards-using-an-incoming-webhook
+func NewAdaptiveCardsMessage(card AdaptiveCard) AdaptiveCardsMessage {
+	return AdaptiveCardsMessage{
+		Attachments: []AdaptiveCardsAttachment{{
+			ContentType: "application/vnd.microsoft.card.adaptive",
+			Content:     card,
+		}},
+		Type: "message",
+	}
+}
+
+// AdaptiveCardsAttachment contains an adaptive card.
+type AdaptiveCardsAttachment struct {
+	Content     AdaptiveCard `json:"content"`
+	ContentType string       `json:"contentType"`
+	ContentURL  string       `json:"contentUrl,omitempty"`
+}
+
+type AdaptiveCard struct {
```

Comment:
```suggestion
// AdaptiveCard is the top level schema for an adaptive card. See: https://adaptivecards.io/explorer/AdaptiveCard.html
type AdaptiveCard struct {
```

Found a nice schema description, might help in the future.